### PR TITLE
Errors

### PR DESCRIPTION
**Pull request by michael-ciniawsky**
- Log Code Frame on `CssSyntaxError` in handler, `err.message` otherwise
- Change error message for output

### CODE REVIEW

**Comment by michael-ciniawsky** [comment]
Maybe add [code-frame](https://github.com/hhsnopek/code-frame)

@ai Can give me the link to where the code frame code is in postcss please ?

**Comment by ai** [comment]
@michael-ciniawsky https://github.com/postcss/postcss/blob/master/lib/css-syntax-error.es6#L158-L199

**Comment by michael-ciniawsky** [comment]
@ai Thx

**Comment by michael-ciniawsky** [comment]
@RyanZim @ai 

```js
...
  // PostCSS Error
  if (err.name === 'CssSyntaxError') {
    console.log(chalk.red(`\nError: ${err.message}`))
    console.log('\n', err.showSourceCode(), '\n')
  } else {
    // PostCSS Config Error
    console.log(chalk.bold.red(err.message))
    process.exit(1)
  }
```

postcss-cli-err

Maybe remove the boldness 😛

**Comment by ai** [comment]
Good :)

**Comment by michael-ciniawsky** [comment]
@ai Is it possible to strip the path from the `err.message` ? Or custom string manipulation

**Comment by ai** [comment]
@michael-ciniawsky just use `str.replace()` :D.

**Comment by michael-ciniawsky** [comment]
```js
// PostCSS Error
  if (err.name === 'CssSyntaxError') {
    err.message = err.message
      .replace(err.file + ':', '[')
      .replace(/:\s/, '] ')

    console.log(chalk.bold.red(`\n${err.message}`))
    console.log('\n', err.showSourceCode(), '\n')
  } else {
    // PostCSS Config Error
    console.log(chalk.bold.red(err.message))
    process.exit(1)
  }
```

postcss-cli-err

Better 😛

File: index.js [hunk 1]
```diff
@@ -139,21 +142,35 @@ function config () {
...
   }
-  console.error(err)
+
+  // PostCSS Error
+  if (err.name === 'CssSyntaxError') {
+    err.message = err.message
+      .replace(`${err.file}:`, '[')
+      .replace(/:\s/, '] ')
+
+    console.log(chalk.bold.red('\n', err.message))
```
Comment (RyanZim) [inline index.js]: Should be `console.error`

File: index.js [hunk 2]
```diff
@@ -139,21 +142,35 @@ function config () {
...
+    console.log('\n', err.showSourceCode(), '\n')
```
Comment (RyanZim) [inline index.js]: Dito

File: index.js [hunk 3]
```diff
@@ -139,21 +142,35 @@ function config () {
...
+
+    return
+  }
+  // PostCSS Config Error
+  console.log(chalk.bold.red(err.message))
```
Comment (RyanZim) [inline index.js]: Dito

**Comment by RyanZim** [comment]
@michael-ciniawsky Also, I question the wisdom of stripping the path from the error message. The CLI can process multiple files at once, and we need to know which file contains the error.

**Comment by michael-ciniawsky** [comment]
> Also, I question the wisdom of stripping the path from the error message.

@RyanZim I can readd it of course, but the file gets already displayed via `spinner.text()`. Every Code Frame + `err.message` should be displayed directly under the current file being processed.

**Comment by RyanZim** [comment]
> Every Code Frame + err.message should be displayed directly under the current file being processed.

True, but that all falls apart pretty fast when two files are being processed at once. I'd still prefer if you removed it.

File: index.js (hunk 3 above)
Comment (michael-ciniawsky) [inline index.js]: I have a question about process.exit(1). Is it mandatory, bc it  throws ugly npm error + npm-debug.log :), is throw not enough to terminate?

**Comment by michael-ciniawsky** [comment]
I will add tests and update the requested changes. We also should add `console.warn(chalk.bold.yellow(...result.messages.warnings))` in `processCSS`, someone aware of a plugin which uses Result#warnings API?

> True, but that all falls apart pretty fast when two files are being processed at once.

In which cases does that happen ?

>  I'd still prefer if you removed it.

You mean the file pathe here ? 😛

File: index.js (hunk 3 above)
Comment (RyanZim) [inline index.js]: - `throw` will still give you an ugly npm error if the script was run in an npm script.
- `throw` prints a verbose stack trace pointing to the line containing the `throw`.
- Since the `error` function is run inside a `catch`, thrown errors will create an unhandled promise rejection.

TL;DR: Just use `process.exit(1)`

**Comment by RyanZim** [comment]
> In which cases does that happen ?

`Promise.all(files.map(file => processCSS(file, config)))` creates parallelism, since we're doing file I/O. 

> > I'd still prefer if you removed it.
>
> You mean the file pathe here ? :stuck_out_tongue:

Yep, print the file path as part of the error message.

**Comment by RyanZim** [comment]
> We also should add console.warn(chalk.bold.yellow(...result.messages.warnings)) in processCSS, someone aware of a plugin which uses Result#warnings API?

I was thinking of using https://github.com/postcss/postcss-reporter, I'd keep this in a separate PR.

**Comment by michael-ciniawsky** [comment]
The error is written to `stderr` how can I get it to be on the `error` prop instead?

**Comment by RyanZim** [comment]
@michael-ciniawsky I think the actual error message will always be in `stderr`. However, just noticed you're not doing `process.exit(1)` for the syntax error. That will make `error` non-null.

**Comment by michael-ciniawsky** [comment]
Already readded `process.exit(1)` locally. Coverage is also fixed, but asserting the error correctly is a pain in the ass, a bit more time on that please :). BTW with  `process.exit(1)` readded the `Processing ${file}` seems to be always in sync with the error thrown, maybe we can remove the `err.file`, needs more investigation still 😛

**Comment by RyanZim** [comment]
You don't have to assert the whole error message, just do `t.regex(stderr, /sugar-white.sss:L:C: Unnecessary curly bracket/)` (Subtituting `L:C` with line & col number)

**Comment by RyanZim** [comment]
Merged, will fix a few nitpicks myself.
